File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/controller/BlogController.java
package jp.co.pluswing.mysqlsample.sample.controller;

import graphql.com.google.common.collect.Lists;
import jp.co.pluswing.mysqlsample.sample.input.NewComment;
import jp.co.pluswing.mysqlsample.sample.input.NewPost;
import jp.co.pluswing.mysqlsample.sample.input.UpdatePost;
import jp.co.pluswing.mysqlsample.sample.model.Comment;
import jp.co.pluswing.mysqlsample.sample.model.Post;
import jp.co.pluswing.mysqlsample.sample.payload.*;
import jp.co.pluswing.mysqlsample.sample.service.BlogService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.graphql.data.method.annotation.Argument;
import org.springframework.graphql.data.method.annotation.MutationMapping;
import org.springframework.graphql.data.method.annotation.QueryMapping;
import org.springframework.stereotype.Controller;

import java.util.List;

@Controller
public class BlogController {

    @Autowired
    private BlogService service;

    @QueryMapping
    public Post post(@Argument Long id) {
        return service.getPostById(id);
    }

    @QueryMapping
    public List<Post> posts() {
        return service.allPost();
    }

    @QueryMapping
    public List<Post> postsByAuthor(@Argument String author) {
        return service.findPostByAuthor(author);
    }

    @QueryMapping
    public List<Post> postsByCategory(@Argument String category) {
        return service.findPostByCategory(category);
    }

    @QueryMapping
    public List<Comment> comments() {
        return service.allComment();
    }

    @MutationMapping
    public AddPostPayload addPost(@Argument NewPost post) {
        try {
            Post p = service.addPost(post);
            return AddPostPayload.success(p);
        } catch (Exception e) {
            return AddPostPayload.fail();
        }
    }

    @MutationMapping
    public DeletePostPayload deletePost(@Argument Long id) {
        try {
            service.deletePost(id);
            return DeletePostPayload.success();
        } catch (Exception e) {
            return DeletePostPayload.fail();
        }
    }

    @MutationMapping
    public UpdatePostPayload updatePost(@Argument UpdatePost post) {
        try {
            Post p = service.updatePost(post);
            return UpdatePostPayload.success(p);
        } catch (Exception e) {
            return UpdatePostPayload.fail();
        }
    }

    @MutationMapping
    public AddCommentPayload addComment(@Argument NewComment comment) {
        try {
            Comment c = service.addComment(comment);
            return AddCommentPayload.success(c);
        } catch (Exception e) {
            return AddCommentPayload.fail();
        }
    }

    @MutationMapping
    public DeleteCommentPayload deleteComment(@Argument Long id) {
        try {
            service.deleteComment(id);
            return DeleteCommentPayload.success();
        } catch (Exception e) {
            return DeleteCommentPayload.fail();
        }
    }
}

File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/controller/BlogController.kt
package jp.co.pluswing.mysqlsample.sample.controller

import jp.co.pluswing.mysqlsample.sample.model.Post
import javax.persistence.GeneratedValue
import javax.persistence.Lob
import javax.persistence.JoinColumn
import jp.co.pluswing.mysqlsample.sample.payload.AddPostPayload
import jp.co.pluswing.mysqlsample.sample.payload.AddCommentPayload
import jp.co.pluswing.mysqlsample.sample.payload.DeletePostPayload
import jp.co.pluswing.mysqlsample.sample.payload.UpdatePostPayload
import jp.co.pluswing.mysqlsample.sample.payload.DeleteCommentPayload
import org.springframework.beans.factory.annotation.Autowired
import jp.co.pluswing.mysqlsample.sample.repository.PostRepository
import jp.co.pluswing.mysqlsample.sample.repository.CommentRepository
import jp.co.pluswing.mysqlsample.sample.input.NewPost
import kotlin.Throws
import jp.co.pluswing.mysqlsample.sample.input.UpdatePost
import jp.co.pluswing.mysqlsample.sample.input.NewComment
import jp.co.pluswing.mysqlsample.sample.model.Comment
import jp.co.pluswing.mysqlsample.sample.service.BlogService
import org.springframework.graphql.data.method.annotation.QueryMapping
import org.springframework.graphql.data.method.annotation.MutationMapping
import org.springframework.data.repository.CrudRepository
import org.springframework.boot.autoconfigure.SpringBootApplication
import kotlin.jvm.JvmStatic
import org.springframework.boot.SpringApplication
import org.springframework.graphql.data.method.annotation.Argument
import org.springframework.stereotype.Controller
import java.lang.Exception

@Controller
class BlogController {
    @Autowired
    private val service: BlogService? = null

    @QueryMapping
    fun post(@Argument id: Long): Post {
        return service!!.getPostById(id)
    }

    @QueryMapping
    fun posts(): List<Post> {
        return service!!.allPost()
    }

    @QueryMapping
    fun postsByAuthor(@Argument author: String): List<Post> {
        return service!!.findPostByAuthor(author)
    }

    @QueryMapping
    fun postsByCategory(@Argument category: String): List<Post> {
        return service!!.findPostByCategory(category)
    }

    @QueryMapping
    fun comments(): List<Comment> {
        return service!!.allComment()
    }

    @MutationMapping
    fun addPost(@Argument post: NewPost): AddPostPayload {
        return try {
            val p = service!!.addPost(post)
            AddPostPayload.success(p)
        } catch (e: Exception) {
            AddPostPayload.fail()
        }
    }

    @MutationMapping
    fun deletePost(@Argument id: Long): DeletePostPayload {
        return try {
            service!!.deletePost(id)
            DeletePostPayload.success()
        } catch (e: Exception) {
            DeletePostPayload.fail()
        }
    }

    @MutationMapping
    fun updatePost(@Argument post: UpdatePost): UpdatePostPayload {
        return try {
            val p = service!!.updatePost(post)
            UpdatePostPayload.success(p)
        } catch (e: Exception) {
            UpdatePostPayload.fail()
        }
    }

    @MutationMapping
    fun addComment(@Argument comment: NewComment): AddCommentPayload {
        return try {
            val c = service!!.addComment(comment)
            AddCommentPayload.success(c)
        } catch (e: Exception) {
            AddCommentPayload.fail()
        }
    }

    @MutationMapping
    fun deleteComment(@Argument id: Long): DeleteCommentPayload {
        return try {
            service!!.deleteComment(id)
            DeleteCommentPayload.success()
        } catch (e: Exception) {
            DeleteCommentPayload.fail()
        }
    }
}